web/components/editor/embed-modal.tsx

```diff
@@ -81,18 +81,41 @@ const embedPatterns: EmbedPattern[] = [
     rewrite: (id) =>
       `<iframe src="https://www.tiktok.com/embed/v2/${id}"></iframe>`,
   },
-  // {
-  //   regex: /^(https?:\/\/.*)/,
-  //   rewrite: (url) => `<iframe src="${url}"></iframe>`,
-  // },
 ]
+const allowed_domains: string[] = [
+  "streamlit.app",
+  "wikipedia.org",
+];
+
+function isAllowedDomain(url: string) {
+  try {
+    const { hostname } = new URL(url);
+    return allowed_domains.some(allowedDomain => 
+      hostname === allowedDomain || hostname.endsWith('.' + allowedDomain)
+    );
+  } catch (error) {
+    return false;
+  }
+}
 
 function embedCode(text: string) {
   for (const pattern of embedPatterns) {
     const match = text.match(pattern.regex)
     if (match) {
       return pattern.rewrite(match[1])
     }
+    
+  }
+
+  const urlPattern = /^(https?:\/\/.*)/;
+  const match = text.match(urlPattern);
+  if (match) {
```

Comment:
Would simplify this to
```
// If it's a URL, check against the allowed domains
if (match && isAllowedDomain(match[1])) {
  return ...
}
```
and exclude the extra return null, because that's already caught by the next case below